backend/src/tools/summaries.py

```diff
@@ -1,7 +1,48 @@
 import langchain_core.tools as lcct
+import google.generativeai as genai
```

Comment:
why are you not using langchain google implementation? Take a look at my `test_graph_builder.test_gemini_model_calls_tool` for an example